Vagrantfile

```diff
@@ -0,0 +1,22 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# All Vagrant configuration is done below. The "2" in Vagrant.configure
+# configures the configuration version (we support older styles for
+# backwards compatibility). Please don't change it unless you know what
+# you're doing.
+Vagrant.configure(2) do |config|
+  config.vm.box = "ubuntu/trusty64"
+  config.vm.network "forwarded_port", guest: 8123, host: 8123
+  config.vm.synced_folder ".", "/vagrant"
+  config.vm.provision "shell", inline: <<-SHELL
```

Comment:
Easier to show you what I mean. With this vagrant file:
https://github.com/shaftoe/home-assistant/commit/9355aea500c4b64f6ecf457f59a894cb6862309b
the workflow could be as it follows:
- clone home-assistant official repo (or, better, a forked one)
- cd into home-assistant/vagrant
- run `vagrant up`

This will take care of installing all the needed stuff to have latest dev version of Home Assistant able to run (in a virtualenv) and bind on local port 8123, plus run the test suite.

At each subsequent `vagrant provision`, it will run tests against the local version of the code.

This is already doing a lot but it's missing some key features, like start/restart hass, that could for example be implemented dropping files into `vagrant/` folder and read their name / content inside the provisioner script to decide what to do when `provision` command is run